src/main.js

```diff
@@ -1,96 +1,148 @@
-import logger from '@financial-times/n-logger';
-
-const MASK_SEQUENCE = '*****';
-
-export default class {
-
-	constructor (sensitiveFields = []) {
-		const fieldsToMask = sensitiveFields.join('|');
-		this.sensitiveFields = new RegExp(`(${fieldsToMask})[\\\s]*\\=[\\\s]*[\\\]?[\\"\\']?([\\S]+)[\\\]?[\\"\\']?|(${fieldsToMask})`, 'ig');
+const logger = require('@financial-times/n-logger').default;
+
+/**
+ * MaskLogger Class
+ * @example
+ * const logger = new MaskLogger(['denyField'], ['allowField'], '*MASKED*');
+ * logger.debug('Data which may contain sensitive information', data);
+ */
+class MaskLogger {
+	/**
+	 * Create a MaskLogger
+	 * @param {array<string>} denyList Extra field names to mask
+	 * @param {array<string>} allowList Field names to allow from the default deny list
+	 * @param {string?} maskString Mask to be applied to sensitive values
+	 */
+	constructor (denyList = [], allowList = [], maskString = '*****') {
+		this.maskString = maskString;
+		this.maskList = [
+			'email',
+			'password',
+			'name',
+			'firstName',
```

Comment:
I was thinking more a greedy regex so to have just have `name` here and will match `name, surname, surName, firstName, secondName, fullName` and then people can whitelist some fields if they really want to bypass or check. Let maybe discuss it if is not clear.

Same for `phone` should match we everything like `telephone` or `mobilePhone`...make sense?

Reply:
The more complex our regex gets the more difficult the class is to understand. Also if we do make these extensive they may capture things that we don't want, such as `offerName`

Reply:
> The more complex our regex gets the more difficult the matching logic is to understand
You create the regex dynamically, I don't think it will be very complex to understand. In term of performances probably in this case regex or not wouldn't change a lot so we could potentially decide to not use the regex at all.
> Also if we do make these extensive they may capture things that we don't want, such as offerName
I think it is better that we mask too much rather than too little for PII. The fact that you will be forced to consciously whitelist fields it will create more awareness of the issue and consciousness of the problem reducing the risk of `human error`.

Reply:
> You create the regex dynamically, I don't think it will be very complex to understand. In term of performances probably in this case regex or not wouldn't change a lot so we could potentially decide to not use the regex at all.

We won't be able to create it dynamically to ensure `name` captures `firstName` `lastName` but not `offerName` this will have to be a very manually crafted regex which will be difficult to understand.

> I think it is better that we mask too much rather than too little for PII. The fact that you will be forced to consciously whitelist fields it will create more awareness of the issue and consciousness of the problem reducing the risk of human error.

I disagree, if your masking is too aggressive it then makes the logs useless, for example we were masking everything with `*email*` which meant it was masking a lot of our URLs meaning we could not see where the logged errors were happening.